GAE/src/org/akvo/flow/servlet/ReportServlet.java

```diff
@@ -0,0 +1,237 @@
+/*
+ *  Copyright (C) 2018 Stichting Akvo (Akvo Foundation)
+ *
+ *  This file is part of Akvo FLOW.
+ *
+ *  Akvo FLOW is free software: you can redistribute it and modify it under the terms of
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.flow.servlet;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Date;
+import java.util.logging.Logger;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.akvo.flow.dao.ReportDao;
+import org.akvo.flow.domain.persistent.Report;
+import org.akvo.flow.rest.dto.ReportTaskRequest;
+import org.codehaus.jackson.JsonGenerationException;
+import org.codehaus.jackson.map.JsonMappingException;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.jfree.util.Log;
+import org.waterforpeople.mapping.app.web.dto.TaskRequest;
+
+import com.gallatinsystems.common.Constants;
+import com.gallatinsystems.common.util.PropertyUtil;
+import com.gallatinsystems.framework.rest.AbstractRestApiServlet;
+import com.gallatinsystems.framework.rest.RestRequest;
+import com.gallatinsystems.framework.rest.RestResponse;
+import com.gallatinsystems.user.dao.UserDao;
+import com.google.appengine.api.taskqueue.Queue;
+import com.google.appengine.api.taskqueue.QueueFactory;
+import com.google.appengine.api.taskqueue.TaskOptions;
+
+
+public class ReportServlet extends AbstractRestApiServlet {
+    private static final Logger log = Logger.getLogger(ReportServlet.class.getName());
+
+    private static final long serialVersionUID = -9064136799930675167L;
+    private static final String SERVLET_URL = "/app_worker/reportservlet";
+
+    private ReportDao rDao;
+    private UserDao uDao;
+
+    class ReportOptions implements Serializable {
+        /**
```

Comment:
This part looks a little strange with the empty comments.

Reply:
fixed